graphql_compiler/compiler/ir_lowering_common/common.py

```diff
@@ -204,7 +210,15 @@ def visitor_fn(expression):
     return new_ir_blocks
 
 
-def extract_folds_from_ir_blocks(ir_blocks):
+def extract_folds_from_ir_blocks(
+    ir_blocks: List[BasicBlockT],
+) -> Tuple[
+    Union[
+        Dict[FoldScopeLocation, List[BasicBlockT]],
+        Dict[FoldScopeLocation, List[Union[CoerceType, Filter, MarkLocation]]],
+    ],
```

Comment:
This is a common thing the auto-generated type hints make, and I don't think it's correct here — it's overly specific.  `Dict[FoldScopeLocation, List[BasicBlockT]]` should be the first thing in the tuple, and not a union.